### PR TITLE
Create CITATION.cff

### PR DESCRIPTION
**Pull request by dirkgr**

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  -
+    affiliation: "Allen Institute for Artificial Intelligence"
+    family-names: Zettlemoyer
+    given-names: Luke
+cff-version: "1.1.0"
+license: "Apache-2.0"
+message: "If you use this software, please cite it using this metadata."
+repository-code: "https://github.com/allenai/allennlp"
+title: AllenNLP
+version: "2.6.0"
```
Comment (AkshitaB) [inline CITATION.cff:61]: This will need to be updated by hand each time we release? We should add a note in our `RELEASE_PROCESS.md`

**Comment by schmmd** [comment]
Don't we want people to cite the following for tracking purposes?

```
@inproceedings{Gardner2017AllenNLP,
  title={AllenNLP: A Deep Semantic Natural Language Processing Platform},
  author={Matt Gardner and Joel Grus and Mark Neumann and Oyvind Tafjord
    and Pradeep Dasigi and Nelson F. Liu and Matthew Peters and
    Michael Schmitz and Luke S. Zettlemoyer},
  year={2017},
  Eprint = {arXiv:1803.07640},
}
```

**Comment by dirkgr** [comment]
We can't reference that in the `CITATION.cff` file though.

**Comment by nelson-liu** [comment]
i believe you can reference the acl workshop paper https://aclanthology.org/W18-2501/ by its DOI ([`10.18653/v1/W18-2501`](https://doi.org/10.18653/v1/W18-2501)) , which can be referenced in a citation.cff file via the `doi` key: https://github.com/citation-file-format/citation-file-format#software-citation-metadata-required

**Comment by schmmd** [comment]
Thanks @nelson-liu I saw that in GitHub's example too.  But overall, what a headache--it's so simple: https://github.com/citation-file-format/citation-file-format

**Comment by dirkgr** [comment]
I've made it match the information from the paper, including the DOI. But I don't know if the information will come out in such a way that putting that citation into your paper leads to a citation in Semantic Scholar.

**Comment by dirkgr** [comment]
Oh, actually I do know. It comes out like this:
```
@misc{Gardner_AllenNLP_A_Deep,
author = {Gardner, Matt and Grus, Joel and Neumann, Mark and Tafjord, Oyvind and Dasigi, Pradeep and Liu, Nelson and Peters, Matthew and Schmitz, Michael and Zettlemoyer, Luke},
doi = {10.18653/v1/W18-2501},
title = {AllenNLP: A Deep Semantic Natural Language Processing Platform},
url = {https://github.com/allenai/allennlp}
}
```
Close enough I guess?

File: CITATION.cff (hunk 1 above)
Comment (dirkgr) [reply CITATION.cff:45]: Done

**Review by AkshitaB — APPROVED** [review]

**Pull request merged**